CLIENT/client_app.py

```
@@ -0,0 +1,65 @@
+
+@dataclass
+class Status:
+    status: str
+    filename: str
+    timestamp: datetime
+    explanation: str
+
+    def is_done(self):
+        return self.status == 'done'
+
+class WebAppClient:
+    def __init__(self, base_url):
+        self.base_url = base_url
+
+    def upload(self, file_path):
+        upload_url = f"{self.base_url}/upload"
+        try:
+            with open(file_path, 'rb') as file:
+                response = requests.post(upload_url, files={'file': file})
+                response.raise_for_status()
+                data = response.json()
+                return data['uid']
+        except requests.exceptions.RequestException as e:
+            raise Exception(f"Upload failed: {e}")
+
+    def status(self, uid):
+        status_url = f"{self.base_url}/status/{uid}"
+        try:
+            response = requests.get(status_url)
+            response.raise_for_status()
+            data = response.json()
+            timestamp = datetime.utcfromtimestamp(data['timestamp'])
+            return Status(data['status'], data['filename'], timestamp, data['explanation'])
+        except requests.exceptions.RequestException as e:
+            raise Exception(f"Status retrieval failed: {e}")
+
+# Example usage
+if __name__ == "__main__":
+    BASE_URL = "http://127.0.0.1:5000"
+    client = WebAppClient(BASE_URL)
+
+    try:
+        # Upload a file
+        file_path = "C:/targilim/Ruty-Presentation.pptx"
+        uid = client.upload(file_path)
+        print(f"Uploaded successfully. UID: {uid}")
+
+
+        # Check status
+        status_obj = client.status(uid)
+        if status_obj.is_done():
+            print("Processing is done!")
+            print(f"Filename: {status_obj.filename}")
+            print(f"Timestamp: {status_obj.timestamp}")
+            print(f"Explanation: {status_obj.explanation}")
+        else:
+            print("Processing is still pending.")
+    except Exception as e:
+        print(f"An error occurred: {e}")
```

In these situations, its better to just throw the exception instead of catching it and printing it